### DISCUSSION TITLE
Prototype firmware for DPFS in BF2

### DISCUSSION THREAD

**Discussion by MightyRaymond** [IBM/DPFS#29] [General]
Hello, Currently we are configuring our own environment to evaluate DPFS by following the instructions in the github ([DPFS/README.md at master · IBM/DPFS · GitHub](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIBM%2FDPFS%2Fblob%2Fmaster%2FREADME.md&data=05%7C01%7Crawang%40microsoft.com%7C4faeeb70b51244b9fb9108dbe1ad727a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638351907294131596%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=uIHTwHO0tNMUnzAqfQP%2BTN2VOPBXohSyQPQEYpAdV5U%3D&reserved=0)) .  From the document, a prototype firmware is required for the whole system to work properly.  Would you please provide more details about this firmware?  Is it developed by NVDIA specifically for the DPFS project?  How can we obtain the binary? Thank you.

**Comment by Peter-JanGootzen** [comment]
Apologies for the ignore, I did not have email notifications set up yet.

So you need two special pieces to get DPFS to work that we sadly cannot help you with.
* The source code of the NVIDIA SNAP library, this is part of DOCA and under NDA.
* A special firmware for the BlueField that has the `virtio-fs` feature enabled. As the feature is currently not generally available yet.

You will have to talk to your Nvidia contact person for these two pieces. Doing this will also help push for `virtio-fs` to become generally available on the BF2. :)
When you have those two pieces, I can send you the async patch set for SNAP over email.
